### PR TITLE
[Annotation] Replace use of id by data-element-id to have the correct id (bug 1796544)

### LINKED ISSUE #15597: Cannot check a checkbox
Attach (recommended) or Link to PDF file here:
[t2210-tax-cat2.pdf](https://github.com/mozilla/pdf.js/files/9824222/t2210-tax-cat2.pdf)

Configuration:
- Web browser and its version: Firefox nightly
- Operating system and its version: Windows 11

Steps to reproduce the problem:
1. Try to check to first checkbox, i.e. Single

What is the expected behavior? (add screenshot)
To have a check mark.

What went wrong? (add screenshot)
Nothing happens.

### PR DESCRIPTION
**Pull request by calixteman**
Closes #15597.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
Given that this looks like a regression, probably from PR #15057, would it be feasible to add an integration-test to ensure that this doesn't accidentally break again?
This particular code-path isn't taken for most forms, so we probably don't have good test-coverage as-is.

File: test/integration/annotation_spec.js [hunk 1]
```diff
@@ -95,6 +95,31 @@ describe("Checkbox annotation", () => {
       );
     });
   });
+
+  fdescribe("issue15597.pdf", () => {
```
Comment (Snuffleupagus) [inline test/integration/annotation_spec.js:99]: Typo?

Comment (calixteman) [reply test/integration/annotation_spec.js:99]: No, oblivion (it's to run only this test, iirc f==focus)

**Pull request merged**